src/React/Twig/ReactComponentExtension.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\React\Twig;
+
+use Symfony\WebpackEncoreBundle\Twig\StimulusTwigExtension;
+use Twig\Environment;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+/**
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * @final
+ * @experimental
+ */
+class ReactComponentExtension extends AbstractExtension
+{
+    private $stimulusExtension;
+
+    public function __construct(StimulusTwigExtension $stimulusExtension)
+    {
+        $this->stimulusExtension = $stimulusExtension;
+    }
+
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('react_component', [$this, 'renderReactComponent'], ['needs_environment' => true, 'is_safe' => ['html_attr']]),
+        ];
+    }
+
+    public function renderReactComponent(Environment $env, string $componentName, array $props = []): string
+    {
+        $props['key'] = md5($componentName.random_bytes(8));
```

Comment:
I assume you're trying to solve the problem of a component being rendered inside of a list? If so, isn't this as bad as *not* specifying the key (except that you have suppressed the warning)? Or is the thinking that... this doesn't really apply to us, because we can't have a parent React component - e.g. `CommentList` that renders (and re-renders) a collection of children e.g. `Comment`?

If so, then it makes sense. But let's only set this if `key` does *not* exist already... just to be safe.